lighthouse-viewer/test/drag-and-drop-test.js

```diff
@@ -29,31 +31,59 @@ describe('DragAndDrop', () => {
 
   afterEach(testHelpers.cleanupJsDomGlobals);
 
-  // TODO: test drop event on document. Callback is not getting called
-  // because jsdom doesn't support clipboard API: https://github.com/tmpvar/jsdom/issues/1568/.
-  it.skip('document responds to drag and drop events', done => {
-    const callback = _ => {
-      assert.ok(true, 'file change callback is called after drop event');
-      done();
-    };
+  it('document responds to drop event with file', () => {
+    const mockCallback = jest.fn();
+    new DragAndDrop(mockCallback);
+
+    // create custom drop event with mock files in dataTransfer
+    document.dispatchEvent(createCustomEvent('drop', 'dataTransfer', {
+      files: ['mock file'],
+    }));
+    expect(mockCallback).toBeCalledWith('mock file');
+  });
+
+  it('document responds to drop event without file', () => {
+    const mockCallback = jest.fn();
+    new DragAndDrop(mockCallback);
+
+    document.dispatchEvent(createCustomEvent('drop', 'dataTransfer', null));
+    expect(mockCallback).not.toBeCalled();
+  });
+
+  it('document responds to dragover event with file', () => {
+    const mockCallback = jest.fn();
+    new DragAndDrop(mockCallback);
+
+    const dragoverEvent = createCustomEvent('dragover', 'dataTransfer', {
+      files: ['mock file'],
+    });
+    document.dispatchEvent(dragoverEvent);
+    expect(dragoverEvent.dataTransfer.dropEffect).toEqual('copy');
+  });
 
-    new DragAndDrop(callback);
+  it('document responds to dragover event without file', () => {
+    const mockCallback = jest.fn();
+    new DragAndDrop(mockCallback);
 
-    document.dispatchEvent(new window.CustomEvent('drop'));
+    const dragoverEvent = createCustomEvent('dragover', 'dataTransfer', null);
+    document.dispatchEvent(dragoverEvent);
+    expect(dragoverEvent.dataTransfer).toBeUndefined();
   });
 
-  it('document responds to drag and drop events', () => {
-    // eslint-disable-next-line no-unused-vars
-    const dragAndDrop = new DragAndDrop();
+  it('document responds to mouseleave event when not dragging', () => {
+    new DragAndDrop(jest.fn);
 
     document.dispatchEvent(new window.CustomEvent('mouseleave'));
```

Comment:
should we align on `createCustomEvent` while we're at it?

Reply:
true, I will made the change align it

Reply:
follow up with https://github.com/GoogleChrome/lighthouse/pull/9314#discussion_r299229402